### ISSUE TITLE
[Task - Other]: Don't store nullish/empty values in the URL

### ISSUE THREAD

**Issue by dpasque** [Automattic/bugomattic#69]
### Details

Right now, the minute you do a history updating action in the site, you get this giant URL!

This is because we are taking all of the tracked state and serializing it in the URL. However, we don't need to serialize all of it all of the time. If things have nullish values (like empty arrays, empty strings, null, undefined), we can just leave that property out of the URL. This will make the URL simpler, and more inviting for people to punch values in or bookmark, etc.

In short, everything in the URL will be meaningful!

### Checklist

- [ ] Update the serializer to exclude nullish/empty values
- [ ] Make sure all the parsers in the reducers handle these okay!
